tests/src/test/java/org/eclipse/hono/tests/registry/TenantAmqpIT.java

```diff
@@ -168,13 +208,35 @@ public void testGetByCaSucceeds(final TestContext ctx) {
     @Test
     public void testGetByCaFailsIfNotAuthorized(final TestContext ctx) {
 
-        final X500Principal subjectDn = new X500Principal("CN=ca-http, OU=Hono, O=Eclipse");
-        tenantClient
-                .get(subjectDn)
+        final String tenantId = helper.getRandomTenantId();
+        final X500Principal subjectDn = new X500Principal("CN=ca-http,OU=Hono,O=Eclipse");
+        final TenantObject payload = new TenantObject()
+                .setTenantId(tenantId)
+                .setTrustAnchor(getRandomPublicKey(), subjectDn)
+                .addAdapterConfiguration(new JsonObject()
+                        .put(TenantConstants.FIELD_ADAPTERS_TYPE, Constants.PROTOCOL_ADAPTER_TYPE_HTTP)
+                        .put(TenantConstants.FIELD_ENABLED, true)
+                        .put(TenantConstants.FIELD_ADAPTERS_DEVICE_AUTHENTICATION_REQUIRED, true));
+
+        helper.registry.addTenant(JsonObject.mapFrom(payload))
+                .compose(r ->  tenantClient.get(subjectDn))
                 .setHandler(ctx.asyncAssertFailure(t -> {
-                    ctx.assertEquals(
-                            HttpURLConnection.HTTP_FORBIDDEN,
-                            ((ServiceInvocationException) t).getErrorCode());
-                }));
+                        ctx.assertEquals(
+                                HttpURLConnection.HTTP_FORBIDDEN,
+                                ((ServiceInvocationException) t).getErrorCode());
+                    }));
+    }
+
+    private PublicKey getRandomPublicKey(){
+        try {
+            final KeyPairGenerator keyGen = KeyPairGenerator.getInstance("RSA");
+            keyGen.initialize(1024);
+            final KeyPair keypair = keyGen.genKeyPair();
+            return keypair.getPublic();
+        } catch (NoSuchAlgorithmException e){
```

Comment:
Why not just propagate the exception from this method, so that test fails if it cannot generate PK?

Reply:
Good suggestion !